cli/src/semgrep/engine.py

```diff
@@ -33,52 +33,54 @@ class EngineType(Enum):
     @classmethod
     def decide_engine_type(
         cls,
-        requested_engine: Optional["EngineType"] = None,
+        logged_in: bool = False,
+        engine_flag: Optional["EngineType"] = None,
+        run_secrets: bool = False,
+        interfile_diff_scan_enabled: bool = False,
+        # ci-only args
         scan_handler: Optional[ScanHandler] = None,
         git_meta: Optional[GitMeta] = None,
-        run_secrets: bool = False,
-        enable_pro_diff_scan: bool = False,
         supply_chain_only: bool = False,
     ) -> "EngineType":
         """Select which Semgrep engine type to use if none is explicitly requested.
 
         Considers settings from Semgrep Cloud Platform and version control state.
         """
-        # Change default to pro-engine intrafile if secrets was requested.
-        # Secrets is built into pro-engine, but any pro-setting should work.
-        if (
-            not (scan_handler and scan_handler.deepsemgrep)
-            and requested_engine is None
-            and run_secrets
-        ):
-            requested_engine = cls.PRO_LANG
-        elif run_secrets and requested_engine is cls.OSS:
-            # Should be impossible if the CLI gates impossible arguement combinations.
-            raise SemgrepError("Semgrep Secrets is not part of the open source engine")
+        interfile_is_requested_via_app = scan_handler and scan_handler.deepsemgrep
 
-        if git_meta and scan_handler:
-            if scan_handler.deepsemgrep and requested_engine is None:
+        requested_engine = engine_flag
```

Comment:
Why not just rename either the parameter of the variable ?

Reply:
Maybe it would be more clear this way:

```python
if engine_flag is None:
  ...
else:
  requested_engine = engine_flag
```

Reply:
Actually I would even put more of the subsequent stuff under the `else:` because then it's more clear that it only applies when `engine_flag` is set.

Reply:
Ok, how do you feel about it now?